src/components/text-search/search-as-you-type.js

```diff
@@ -0,0 +1,344 @@
+  // on current state.
+  handleSubmit() {
+    this.props.onChange(this.props.field, this.state.value);
+    // Get existing querystring params.
+    const parsed = queryString.parse(window.location.search);
+    // Update the search querystring param with the value from the search field.
+    parsed.search = this.state.value;
+    const stringified = queryString.stringify(parsed);
+    // Update the querystring params in the browser, add path to history.
+    // See: https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_pushState()_method
+    if (window.history.pushState) {
+      const newUrl = `${window.location.protocol}//${window.location.host}${window.location.pathname}?${stringified}`;
+      window.history.pushState({ path: newUrl }, '', newUrl);
+    } else {
+      window.location.search = stringified;
+    }
+  }
+
+  // When the input is focused, Autosuggest will consult this function
+  // when to render suggestions. Use it, for example, if you want to
+  // display suggestions when input value is at least 2 characters long.
+  shouldRenderSuggestions(value) {
+    const numChars = this.props.autocomplete.numChars || 2;
+    return value.trim().length > numChars;
+  }
+
+  renderSuggestionsContainer({ containerProps, children, query }) {
+    const { mode } = this.props.autocomplete;
+    const titleText = this.props.autocomplete[mode].titleText || 'What are you looking for?';
+    const directionsText = this.props.autocomplete[mode].directionsText || true;
+    const suggestionsWrapperClasses = directionsText
+      ? 'react-autosuggest__suggestions-itemslist-wrapper react-autosuggest__suggestions-itemslist-wrapper--with-directions'
+      : 'react-autosuggest__suggestions-itemslist-wrapper';
+
+    return (
+      <div {... containerProps}>
+        <div className="react-autosuggest__container-title">
+          {titleText}
+          <button className="react-autosuggest__container-close-button" onClick={this.onSuggestionsClearRequested}>x</button>
+        </div>
+        <div className={suggestionsWrapperClasses}>
+          {children}
+        </div>
+        {/* @todo add logic for suggestion mode and alter directionsText accordingly */}
+        {directionsText &&
+          <div className="react-autosuggest__container-directions">
+            <span className="react-autosuggest__container-directions-item">Press <code>ENTER</code> to search for <strong>{query}</strong> or <code>ESC</code> to close.</span>
+            <span className="react-autosuggest__container-directions-item">Press ↑ and ↓ to highlight a suggestion then <code>ENTER</code> to be redirected to that suggestion.</span>
+          </div>
+        }
+      </div>
+    );
+  }
+
+  /**
+   * Define how suggestions are rendered.
+   * Note: must be a pure function.
+   *
+   * @param suggestion
+   *   The suggestion to render
+   *
+   * @param query
+   * Used to highlight the matching string. As user types in the input,
```

Comment:
Indent?

Reply:
Yep. Fixed in https://github.com/palantirnet/federated-search-react/pull/26/commits/85fcd96a25e5fde0b32e8a17f4b4813bc222e25f